### PR TITLE
Fixes #35207 - dual stack fallback

### PR DESCRIPTION
**Pull request by jcpunk**
This should let folks select which stack they want to provision over clearly with an easy way to disable it once dual stack works.

### CODE REVIEW

File: app/views/unattended/provisioning_templates/snippet/kickstart_kernel_options.erb [hunk 1]
```diff
@@ -52,7 +53,15 @@ description: |
   end
 
   # networking credentials
-  raise("Dual-stack provisioning not supported") if subnet4 && subnet6
+  if subnet4 && subnet6
+    if dual_stack_fallback == 'IPv4'
+      subnet6 = false
+    elsif dual_stack_fallback == 'IPv6'
+      subnet = false
```
Comment (evgeni) [inline app/views/unattended/provisioning_templates/snippet/kickstart_kernel_options.erb]:
```suggestion
      subnet4 = false
```

Or is there also a subnet param that I'm not seeing right now?

**Comment by evgeni** [comment]
@ares @lzap does one of you have a setup to test this on? It seems to make sense to me, but I have no way to test

File: app/views/unattended/provisioning_templates/snippet/kickstart_kernel_options.erb (hunk 1 above)
Comment (jcpunk) [reply app/views/unattended/provisioning_templates/snippet/kickstart_kernel_options.erb]: That was a typo :)

**Comment by lzap** [comment]
I almost never did test these edge cases, I would only make sure the generated files are valid. These are not supported in Satellite so we are usually good to go just by verifying the syntax. Just keep in mind that our unit tests (ksvalidator) snapshot suite does only basic configuraiton (single IPv4, single IPv6), if you want to go an extra mile, just request adding a test case there.

LGTM

**Review by lzap — APPROVED** [review]
Okay, but please keep in mind some Foreman features are hard to work with with dualstack as Foreman DB/UI is not ready to have two subnets associated. We very often do:

if subnet6 do this, else if subnet4 do this.
